pallene/Lexer.lua

```diff
@@ -266,8 +267,19 @@ function Lexer:next()
         if not name then
             return false, val
         end
-        if name ~= "SPACE" and name ~= "COMMENT" then
-            return { name = name, value = val, loc = loc }
+
+        local end_loc = self:loc()
+        if name == "COMMENT" then
+            -- For a comment, `end_loc.pos` points at the starting position of the next token.
+            -- Therefore, we subtract 1 from the position to adjust this anamoly.
```

Comment:
I think we can get rid of this comment. This isn't an anomaly. `self:loc()` points to the start of the next token, meaning that `end_loc.pos` pointing to one character past the end of the comment is to be expected.